spec/solver_spec.rb

```diff
@@ -0,0 +1,49 @@
+require_relative '../solver'
+
+describe Solver do
+  before :all do
+    @solver = Solver.new
+  end
+
+  context 'factorial method' do
+    it 'should compute factorial for non-zero positive number' do
+      expect(@solver.factorial(5)).to eq(120)
+    end
+
+    it 'should return 1 for number 0' do
+      expect(@solver.factorial(0)).to eq(1)
+    end
+
+    it 'should raise en error for negative number' do
+      expect { @solver.factorial(-1) }.to raise_error(TypeError)
+    end
+  end
```

Comment:
- [ ] If the value is negative it should raise an exception 
![image](https://user-images.githubusercontent.com/62705947/218067256-093feefb-cd12-4342-9db0-287e744aea28.png)

Reply:
Hi @GzimAsani, I don't understand the method is raising an error?

Reply:
Hi @Abdessittir, great work. I guess the previous reviewer was trying to say that you should raise an exception with an intuitive message as the parameter. Something like this:

```ruby
if num.negative?
  raise TypeError, 'Negative number'
```